### PR TITLE
[wasm][tests] Add WebSocket tests

### PR DESCRIPTION
**Pull request by kjpou1**
Add tests for websocket interface.

Includes:
Fix issue https://github.com/mono/mono/issues/13438

### CODE REVIEW

**Comment by kjpou1** [comment]
@monojenkins build failed

**Comment by kjpou1** [comment]
@monojenkins build Linux WebAssembly

File: sdks/wasm/tests/browser/WebSocketTestSuite.cs [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+			}
+
+			return cws;
+
+		}
+		public async Task<int> ConnectWebSocketStatus (Uri server, string protocols)
+		{
+			var cws = CreateWebSocket (server, protocols);
+
+            WebSocketCloseStatus status = cws.CloseStatus ?? WebSocketCloseStatus.Empty;
```
Comment (kg) [inline sdks/wasm/tests/browser/WebSocketTestSuite.cs]: indentation damage all over in this file, fyi

Comment (kjpou1) [reply sdks/wasm/tests/browser/WebSocketTestSuite.cs]: should be fixed in recent commit.  Totally forgot to check format.

**Comment by kjpou1** [comment]
@monojenkins build Failed

**Comment by kjpou1** [comment]
@monojenkins build Failed

**Comment by kjpou1** [comment]
@monojenkins build Linux WebAssembly

**Comment by kjpou1** [comment]
@monojenkins build failed

**Comment by kjpou1** [comment]
@monojenkins build failed

**Comment by kjpou1** [comment]
@monojenkins build Linux WebAssembly

**Comment by kjpou1** [comment]
@monojenkins build Linux WebAssembly

**Comment by kjpou1** [comment]
@monojenkins build Linux WebAssembly

**Pull request merged**